src/server.c

```diff
@@ -2039,6 +2039,7 @@ void initServerConfig(void) {
                                       updated later after loading the config.
                                       This value may be used before the server
                                       is initialized. */
+    server.io_threads_num = server.config_io_threads_num;
```

Comment:
this line probably doesn't belong here.
and i'm not certain about the one in initServer either....
ideally, we can set `server.io_threads_num` to 0 or 1, and then call the update function at startup to match the thread count to the config.